src/function.ts

```diff
@@ -42,11 +43,12 @@ export class RustFunction extends lambda.Function {
     const manifestPath = getCargoManifestPath(props?.manifestPath ?? 'Cargo.toml');
 
     const runtime = lambda.Runtime.PROVIDED_AL2;
-    const bundling = props?.bundling ?? {};
+    const bundling = bundlingOptionsFromRustFunctionProps(props);
 
     super(scope, packageName, {
       ...props,
       runtime,
+      architecture: bundling.architecture,
```

Comment:
For better testability I set the architecture explicitly. Without that the generated CloudFormation does not contain the `Architectures` property. So I guess CF decides at some point to use `x86_64` during deployment.